### PR TITLE
Vita49 to sigmf converter

### PR DESCRIPTION
**Pull request by SimonWittkowski**
Current status of a Vita49 to SigMF converter. It is still WIP, and will be continued in july 2024. Due to the lack of test data it is currently only tested with self generated vita49 testdata (validated with Matlab vita49 reader).
Change the path in vita49_file_reader to your vita49 file and run vita49_file_reader to convert your vita49 data to sigMF. Converting DIFI packets should be possible as well as DIFI is Vita compliant.

### CODE REVIEW

File: api/converters/vita49_to_sigmf/vita49.py
Comment (Nepomuceno) [inline api/converters/vita49_to_sigmf/vita49.py:21]: probably typo

**Comment by 777arc** [comment]
@SimonWittkowski got an example vita file that I can use to test with?

**Comment by 777arc** [comment]
@SimonWittkowski  OK I made a ton of changes, none to the actual packet parsing logic though (aside from adding pcap support which i wasnt able to actually test), but now the converter will show up in the Web GUI (https://iqengine.org/convert once we merge and push to prod), but it should also be usable straight from python.  I'm ready to merge whenever but if you wanted to go through and do any cleanups, or add support for situations like when the recording doesnt start with a context packet, feel free.

**Comment by SimonWittkowski** [comment]
> @SimonWittkowski OK I made a ton of changes, none to the actual packet parsing logic though (aside from adding pcap support which i wasnt able to actually test), but now the converter will show up in the Web GUI (https://iqengine.org/convert once we merge and push to prod), but it should also be usable straight from python. I'm ready to merge whenever but if you wanted to go through and do any cleanups, or add support for situations like when the recording doesnt start with a context packet, feel free.

**Comment by 777arc** [comment]
No need to close the PR, you can add commits to the same branch, and just let me know when it's ready to merge

**Comment by SimonWittkowski** [comment]
current problems fixed, ready to be merged

**Review by 777arc — APPROVED** [review]

**Comment by 777arc** [comment]
Nice!!! Merging!

**Pull request merged**
